Brix IPC 783

ysahnpark opened the pull request:
Pull request for code review.

The change includes implementation of IPC. 
I used sinon mocking library in the unit tests.

Commit:
IPSProxy utest and logic

Commit:
Proper comments to ipsproxy.js

Commit:
Minor additions to the comment

ysahnpark committed:
Merge remote-tracking branch 'upstream/BricLayer' into BrixIPC_783

Merged latest BricLayer

ysahnpark committed:
DomHelper's scanElement/Object implemented & tested

ysahnpark committed:
DomHelper's scanElement/Object implemented & tested

ysahnpark committed:
Working on ipc unit test

ysahnpark committed:
Merge remote-tracking branch 'upstream/master' into BrixIPC_783

BrixLayer update merged

ysahnpark committed:
Added mocking framework sinon-1.7.3.js

ysahnpark committed:
Unit test for ipc

ysahnpark committed:
Improved comments

seannives commented on js/domhelper.js:
```diff
@@ -1,19 +1,19 @@
+ * @export
  *
  */
 var DomHelper = function(options) {
```
missing semi on L20?

seannives commented on js/domhelper.js:
```diff
@@ -37,52 +37,77 @@ DomHelper.prototype.frameCustomParams = new Array();
+	for (var i = 0; i < this.framesList.length; i++){
+		if (windowObj === this.framesList[i].contentWindow) {
+			index = i;
+			break;
```
Asking because I don't know...  is it best practice to break here and return outside of the loop rather than put the return inside the loop?

ysahnpark replied:
There's been debate that it is good to have one return statement(s) centralized at the bottom of the function for readability purpose.
But small codes like this one, sometimes makes it easier to read when you return right within the condition.
So, I'll changed it to return within the loop.

seannives commented on js/domhelper.js:
```diff
@@ -37,52 +37,77 @@ DomHelper.prototype.frameCustomParams = new Array();
-
-DomHelper.prototype.getFrameCustomParamsByIndex = function(index) {
+	if (index > -1)
 		return this.frameCustomParams[index];
```
Can we put {} around the return?  It's not necessary but I kind of prefer it...  not sure what the style guide says on this.

seannives commented on js/ipc.js:
```diff
@@ -0,0 +1,219 @@
+     * The IpsProxy instance
+     * @type {pearson.brix.IpsProxy}
+     */
+    this.ipsProxy = new pearson.brix.IpsProxy({"serverBaseUrl":"http://localhost:8088"});
```
We should figure out how to abstract serverBaseUrl out to config.

seannives commented on js/ipc.js:
```diff
@@ -0,0 +1,219 @@
+ * The different parameters defines the mode: div or iframe.
+ * In div-mode, the items contains an array of possibly multiple items, and 
+ * the optional opt_containerId is not defined (or null).
+ * In firame-mode, there is only one single item and opt_containerId is passed
```
typeo: firame-mode to iframe mode

seannives commented on js/ipc.js:
```diff
@@ -0,0 +1,219 @@
+ */
+pearson.brix.Ipc.prototype.init = function(items, opt_containerId)
+{
+    if (!items || items.length == 0)
```
If the type's the same here can it be changed to items.length ===  0?  I've been meaning to be better about this practice myself...

seannives commented on js/ipsproxy.js:
```diff
@@ -0,0 +1,204 @@
+
+
+/* **************************************************************************
+ * Retrieves sequence node. The server is health if the callback is called
```
Do you need the "Retrieves sequence node." sentence?  Was that a copy/paste from elsewhere?

ysahnpark committed:
DomHelper refactored and compiled with closure, other minor fixes after code review

ysahnpark commented:
I run compiler on DomHelper and fixed warnings.
Also refactored DomHelper: the iframe related methods were factored out as IframeCollection and placed inside MessageBroker.
